pkg/registry/apis/provisioning/register.go

```diff
@@ -380,16 +380,22 @@ func (b *APIBuilder) Mutate(ctx context.Context, a admission.Attributes, o admis
 			r.Spec.GitHub.Branch = "main"
 		}
 
+		if r.Spec.Workflows == nil {
+			r.Spec.Workflows = []provisioning.Workflow{
+				provisioning.BranchWorkflow,
+				provisioning.WriteWorkflow,
+			}
+		}
+
 		// Trim trailing slash or .git
 		if len(r.Spec.GitHub.URL) > 5 {
 			r.Spec.GitHub.URL = strings.TrimRight(strings.TrimRight(r.Spec.GitHub.URL, "/"), ".git")
 		}
+	}
 
-		if len(r.Spec.GitHub.Workflows) == 0 {
-			r.Spec.GitHub.Workflows = []provisioning.Workflow{
-				provisioning.BranchWorkflow,
-				provisioning.PushWorkflow,
-			}
+	if r.Spec.Workflows == nil {
```

Comment:
how do people then have this read only? passing an empty list? It seems like less intuitive than before. I guess we should not set defaults for users in the mutator but only in the frontend to help them. The backend will be read-only by default.

Reply:
i added examples in the tests... agree it is not the most obvious, but it does work.  *missing* workflow will mutate and fill in a [branch, write]:
https://github.com/grafana/grafana/blob/workflows-as-write-access/pkg/tests/apis/provisioning/testdata/local-tmp.json

but if defined as empty, it will stay empty -- eg readonly:
https://github.com/grafana/grafana/blob/workflows-as-write-access/pkg/tests/apis/provisioning/testdata/local-xxx.json#L12

Reply:
I do wonder if having an explicit, mutually exclusive read-only workflow might be a good idea as well. That'd make the user have to opt into _some_ mode.

Reply:
it's not very obvious. Very subtle difference. I would still default to read-only if not specified. 

cc @Proximyst